crud/src/test/java/com/redhat/lightblue/rest/crud/ITCaseCrudResourceTest.java

```diff
@@ -317,4 +317,20 @@ public void testLock() throws IOException, ClassNotFoundException, NoSuchMethodE
         String result=cutCrudResource.acquire("test","caller","resource",null).getEntity().toString();
         Assert.assertEquals("{\"result\":true}",result);
     }
+
+    @Test
+    public void testGenerate() throws IOException, ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException, URISyntaxException, JSONException {
+        Assert.assertNotNull("CrudResource was not injected by the container", cutCrudResource);
+        String metadata = readFile("generator-md.json");
+        EntityMetadata em = RestConfiguration.getFactory().getJSONParser().parseEntityMetadata(JsonUtils.json(metadata));
+        RestConfiguration.getFactory().getMetadata().createNewMetadata(em);
+
+        String result=cutCrudResource.generate("generate","1.0.0","number",1).getEntity().toString();
+        System.out.println("Generated:"+result);
+        JSONAssert.assertEquals("{\"processed\":[\"50000000\"]}",result,false);
+
+        result=cutCrudResource.generate("generate","1.0.0","number",3).getEntity().toString();
+        System.out.println("Generated:"+result);
+        JSONAssert.assertEquals("{\"processed\":[\"50000001\",\"50000002\",\"50000003\"]}",result,false);
+    }
```

Comment:
Modified the test to verify that as well


On Wed, May 4, 2016 at 12:45 PM, Naveen Malik <notifications@github.com>
wrote:

> In
> crud/src/test/java/com/redhat/lightblue/rest/crud/ITCaseCrudResourceTest.java
> <https://github.com/lightblue-platform/lightblue-rest/pull/229#discussion_r62092890>
> :
>
> > +
> > +    @Test
> > +    public void testGenerate() throws IOException, ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException, URISyntaxException, JSONException {
> > +        Assert.assertNotNull("CrudResource was not injected by the container", cutCrudResource);
> > +        String metadata = readFile("generator-md.json");
> > +        EntityMetadata em = RestConfiguration.getFactory().getJSONParser().parseEntityMetadata(JsonUtils.json(metadata));
> > +        RestConfiguration.getFactory().getMetadata().createNewMetadata(em);
> > +
> > +        String result=cutCrudResource.generate("generate","1.0.0","number",1).getEntity().toString();
> > +        System.out.println("Generated:"+result);
> > +        JSONAssert.assertEquals("{\"processed\":[\"50000000\"]}",result,false);
> > +
> > +        result=cutCrudResource.generate("generate","1.0.0","number",3).getEntity().toString();
> > +        System.out.println("Generated:"+result);
> > +        JSONAssert.assertEquals("{\"processed\":[\"50000001\",\"50000002\",\"50000003\"]}",result,false);
> > +    }
>
> Test when no version is supplied?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-rest/pull/229/files/0bc69428bb58e29e04c0745b8fdbf722b633a7bc#r62092890>
>